example/src/new_api/swipeable/Swipeable.tsx

```diff
@@ -0,0 +1,606 @@
+   */
+  onSwipeableClose?: (
+    direction: 'left' | 'right',
+    swipeable: ForwardedRef<SwipeableProps & RefAttributes<ExposedFunctions>>
+  ) => void;
+
+  /**
+   * @deprecated Use `direction` argument of onSwipeableWillOpen()
+   *
+   * Called when left action panel starts animating on open.
+   */
+  onSwipeableLeftWillOpen?: () => void;
+
+  /**
+   * @deprecated Use `direction` argument of onSwipeableWillOpen()
+   *
+   * Called when right action panel starts animating on open.
+   */
+  onSwipeableRightWillOpen?: () => void;
+
+  /**
+   * Called when action panel starts animating on open (either right or left).
+   */
+  onSwipeableWillOpen?: (direction: 'left' | 'right') => void;
+
+  /**
+   * Called when action panel starts animating on close.
+   */
+  onSwipeableWillClose?: (direction: 'left' | 'right') => void;
+
+  /**
+   * Called when action panel starts being shown on dragging to open.
+   */
+  onSwipeableOpenStartDrag?: (direction: 'left' | 'right') => void;
+
+  /**
+   * Called when action panel starts being shown on dragging to close.
+   */
+  onSwipeableCloseStartDrag?: (direction: 'left' | 'right') => void;
+
+  /**
+   *
+   * This map describes the values to use as inputRange for extra interpolation:
+   * AnimatedValue: [startValue, endValue]
+   *
+   * progressAnimatedValue: [0, 1] dragAnimatedValue: [0, +]
+   *
+   * To support `rtl` flexbox layouts use `flexDirection` styling.
+   * */
+  renderLeftActions?: (
+    progressAnimatedValue: SharedValue,
+    dragAnimatedValue: SharedValue,
+    swipeable: ForwardedRef<SwipeableProps & RefAttributes<ExposedFunctions>> // we have to use ref here, as it now holds all the objects
+  ) => React.ReactNode;
+  /**
+   *
+   * This map describes the values to use as inputRange for extra interpolation:
+   * AnimatedValue: [startValue, endValue]
+   *
+   * progressAnimatedValue: [0, 1] dragAnimatedValue: [0, -]
+   *
+   * To support `rtl` flexbox layouts use `flexDirection` styling.
+   * */
+  renderRightActions?: (
+    progressAnimatedValue: SharedValue,
+    dragAnimatedValue: SharedValue,
+    swipeable: ForwardedRef<SwipeableProps & RefAttributes<ExposedFunctions>>
+  ) => React.ReactNode;
+
+  useNativeAnimations?: boolean;
+
+  animationOptions?: Record<string, unknown>;
+
+  /**
+   * Style object for the container (`Animated.View`), for example to override
+   * `overflow: 'hidden'`.
+   */
+  containerStyle?: StyleProp<ViewStyle>;
+
+  /**
+   * Style object for the children container (`Animated.View`), for example to
+   * apply `flex: 1`
+   */
+  childrenContainerStyle?: StyleProp<ViewStyle>;
+}
+
+export interface ExposedFunctions {
+  close: () => void;
+  openLeft: () => void;
+  openRight: () => void;
+  reset: () => void;
+}
+
+const Swipeable = forwardRef<
+  SwipeableProps & React.RefAttributes<ExposedFunctions>
+>(function Swipeable(
+  props: SwipeableProps,
+  ref: ForwardedRef<SwipeableProps & RefAttributes<ExposedFunctions>>
+) {
```

Comment:
```suggestion
const Swipeable = forwardRef<ExposedFunctions, SwipeableProps>(
  function Swipeable(
    props: SwipeableProps,
    ref: ForwardedRef<ExposedFunctions>
  ) {
```

Ref methods and props are separate from each other. `forwardRef` allows you to specify both - the first generic type is for ref, the second one is for props.

Reply:
Thank you, this fixed a lot of other related issues.